pkg/logger/logger_test.go

```diff
@@ -0,0 +1,246 @@
+				messages <- text
+			}
+		}(conn)
+	}
+
+	// no more incoming connections, wait for all connections to be closed
+	wg.Wait()
+	close(messages)
+}
+
+func startHTTPServer(
+	ctx context.Context,
+	port string,
+	handler http.Handler,
+	ready chan<- struct{},
+	shutdownHTTP <-chan struct{},
+	shutdownSyslog chan<- struct{},
+) {
+	var wg sync.WaitGroup
+	customHandler := NewCustomHandler(&wg, handler)
+	server := &http.Server{
+		Addr:    ":" + port,
+		Handler: customHandler,
+	}
+
+	// Start the server
+	go func() {
+		if err := server.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+			log.Fatalf("Error starting the server: %v", err)
+		}
+	}()
+
+	// Wait until the server is ready
+	timeout := time.After(5 * time.Second)           // Wait up to 5 seconds for server to start
+	ticker := time.NewTicker(100 * time.Millisecond) // Check every 100ms if server is ready
+	defer ticker.Stop()
+loop:
+	for {
+		select {
+		case <-timeout:
+			log.Fatal("Server start timeout")
+			return
+		case <-ticker.C:
+			conn, err := net.Dial("tcp", ":"+port)
+			if err == nil {
+				_ = conn.Close()
+				ready <- struct{}{}
+				break loop
+			}
+		}
+	}
+
+	// Listen for shutdown signal
+	go func(ctx context.Context) {
+		<-shutdownHTTP
+
+		// Start shutdown process of HTTP server
+		if err := server.Shutdown(ctx); err != nil {
+			log.Fatalf("Server Shutdown: %v", err)
+		}
+
+		// Wait for all HTTP requests to be processed
+		// Feature that we test on the prod server
+		wg.Wait()
+
+		// Notify that the HTTP server is shutdown
+		close(shutdownSyslog)
+	}(ctx)
+}
+
+func TestToto(t *testing.T) {
+	amountOfMessages := 100 // amount of messages to send to the logger
+
+	syslogReady := make(chan struct{})              // channel to notify when the syslog server is ready
+	shutdownSyslog := make(chan struct{})           // channel to notify when to shutdown the syslog server
+	messages := make(chan string, amountOfMessages) // channel to receive syslog messages
+	httpReady := make(chan struct{})                // channel to notify when the http server is ready
+	shutdownHTTP := make(chan struct{})             // channel to notify when to shutdown the http server
+
+	go syslogServer("16901", messages, syslogReady, shutdownSyslog)
+
+	<-syslogReady // Wait until the syslog server is ready
+
+	// start the logger and http server
+	logCfg := Config{
+		Level:     "info",
+		Component: "dagger-registry-2023-07-28",
+		Protocol:  "tcp",
+		Address:   "127.0.0.1:16901",
+	}
+
+	ctx, syslogger, err := NewLogger(context.Background(), &logCfg)
+	if err != nil {
+		panic(err)
+	}
+
+	// init the server mux
+	cancelContext, cancel := context.WithCancel(ctx)
+
+	mux := http.NewServeMux()
+	handler := makeLoggingHandler(cancelContext)
+	mux.HandleFunc("/", handler)
+
+	go startHTTPServer(ctx, "8080", mux, httpReady, shutdownHTTP, shutdownSyslog)
+
+	<-httpReady // Wait until the http server is ready
+
+	// Send 100 requests to the http server
```

Comment:
remove all hard-coded values from comments. if they change (happens more often than you think) the comments will be outdated.